### PR TITLE
Rename systems and change hierarchy

### PR DESCRIPTION
**Pull request by ndido98**
EmptySystem is now System, and System[L] is now IteratingSystem[L].
Now IteratingSystem extends from System and not viceversa.

### CODE REVIEW

File: core/src/test/scala/dev/atedeg/ecscala/SystemTest.scala [hunk 1]
```diff
@@ -8,7 +8,28 @@ import org.scalatest.wordspec.AnyWordSpec
 class SystemTest extends AnyWordSpec with Matchers {
   def beAbleTo = afterWord("be able to")
 
-  "A System" when {
+  "A System" should {
```
Comment (giacomocavalieri) [inline core/src/test/scala/dev/atedeg/ecscala/SystemTest.scala:11]: I don't like the `should` here, maybe change it with a `when`?

File: core/src/test/scala/dev/atedeg/ecscala/fixtures/SystemFixture.scala [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 import dev.atedeg.ecscala.dsl.ECScalaDSL
 import dev.atedeg.ecscala.util.types.given
 
 trait SystemFixture extends ECScalaDSL {
 
-  lazy val mySystem1 = System[Position &: CNil]((_, comps, _) => {
+  lazy val mySystem1 = IteratingSystem[Position &: CNil]((_, comps, _) => {
     val Position(x, y) &: CNil = comps
     Position(x + 3, y + 3) &: CNil
   })
```
Comment (giacomocavalieri) [inline core/src/test/scala/dev/atedeg/ecscala/fixtures/SystemFixture.scala:12]:
```suggestion
 val mySystem1 = IteratingSystem[Position &: CNil] { (_, comps, _) =>
    val Position(x, y) &: CNil = comps
    Position(x + 3, y + 3) &: CNil
  }
```

File: core/src/test/scala/dev/atedeg/ecscala/fixtures/SystemFixture.scala [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
 
-  lazy val mySystem2 = System[Position &: CNil]((_, comps, _) => {
+  lazy val mySystem2 = IteratingSystem[Position &: CNil]((_, comps, _) => {
     val Position(x, y) &: CNil = comps
     Position(x + 1, y + 1) &: CNil
   })
```
Comment (giacomocavalieri) [inline core/src/test/scala/dev/atedeg/ecscala/fixtures/SystemFixture.scala:17]:
```suggestion
  val mySystem2 = IteratingSystem[Position &: CNil] { (_, comps, _) =>
    val Position(x, y) &: CNil = comps
    Position(x + 1, y + 1) &: CNil
  }
```

File: core/src/main/scala/dev/atedeg/ecscala/System.scala [hunk 4]
```diff
@@ -8,63 +8,96 @@ import dev.atedeg.ecscala.util.types.{ taggedWith, CListTag, ComponentTag }
...
    * @param world
    *   the [[World]] in which the [[System]] is being executed.
+   */
+  def update(deltaTime: DeltaTime, world: World): Unit
+}
+
+object System {
+
+  /**
+   * Create a [[System]] from a lambda that specifies its behaviuor.
```
Comment (giacomocavalieri) [inline core/src/main/scala/dev/atedeg/ecscala/System.scala:40]:
```suggestion
   * Create a [[System]] from a lambda that specifies its behaviour.
```

File: core/src/main/scala/dev/atedeg/ecscala/System.scala [hunk 5]
```diff
@@ -8,63 +8,96 @@ import dev.atedeg.ecscala.util.types.{ taggedWith, CListTag, ComponentTag }
...
+   * @param f
+   *   the behaviuor of the [[System]] that takes [[DeltaTime]] and the [[World]] and returns Unit.
```
Comment (giacomocavalieri) [inline core/src/main/scala/dev/atedeg/ecscala/System.scala:42]:
```suggestion
   *   the behaviour of the [[System]] that takes [[DeltaTime]] and the [[World]] and returns Unit.
```

**Review by giacomocavalieri — CHANGES_REQUESTED** [review]

**Review by nicolasfara — APPROVED** [review]

**Review by giacomocavalieri — APPROVED** [review]

**Review by vitlinda — APPROVED** [review]

**Pull request merged**